lib/android/src/main/java/com/reactnativeldk/LdkModule.kt

```diff
@@ -790,11 +794,39 @@ class LdkModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMod
             UserConfig.with_default()
         )
 
-        if (!res.is_ok) {
-            return handleReject(promise, LdkErrors.start_create_channel_fail)
+        if (res.is_ok) {
+            return promise.resolve((res as Result_ChannelIdAPIErrorZ_OK).res._a.hexEncodedString())
+        }
+
+        if ((res as Result_ChannelIdAPIErrorZ.Result_ChannelIdAPIErrorZ_Err).err is APIError.APIMisuseError) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error((res.err as APIError.APIMisuseError).err))
+        }
+
+        if (res.err is APIError.ChannelUnavailable) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error((res.err as APIError.ChannelUnavailable).err))
+        }
+
+        if (res.err is APIError.FeeRateTooHigh) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error((res.err as APIError.FeeRateTooHigh).err))
+        }
+
+        if (res.err is APIError.InvalidRoute) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error((res.err as APIError.InvalidRoute).err))
+        }
+
+        if (res.err is APIError.IncompatibleShutdownScript) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error("IncompatibleShutdownScript"))
+        }
+
+        if (res.err is APIError.MonitorUpdateInProgress) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error("MonitorUpdateInProgress"))
+        }
+
+        if (res.err is APIError.ChannelUnavailable) {
+            return handleReject(promise, LdkErrors.start_create_channel_fail, Error((res.err as APIError.ChannelUnavailable).err))
         }
 
-        handleResolve(promise, LdkCallbackResponses.start_create_channel_fail)
+        return handleReject(promise, LdkErrors.start_create_channel_fail)
```

Comment:
Since we can profit from smart casting here, I'm suggesting -- with a little help from Gemini and my own checks to ensure it compiles -- to go with this instead:
```kt
val failure = when(val error = (res as? Result_ChannelIdAPIErrorZ.Result_ChannelIdAPIErrorZ_Err)?.err) {
    is APIError.APIMisuseError -> Error(error.err)
    is APIError.ChannelUnavailable -> Error(error.err)
    is APIError.FeeRateTooHigh -> Error(error.err)
    is APIError.InvalidRoute -> Error(error.err)
    is APIError.IncompatibleShutdownScript -> Error("IncompatibleShutdownScript")
    is APIError.MonitorUpdateInProgress -> Error("MonitorUpdateInProgress")
    else -> null
}

return handleReject(promise, LdkErrors.start_create_channel_fail, failure)
```

Hope you like it 🤞🏻 , the `error.err` is already "smart-cast" so we don't need to cast again

Reply:
cool updated